ehrql/dummy_data/generator.py

```diff
@@ -209,9 +209,12 @@ def populate_row(self, table_info, row):
 
     def get_random_value(self, column_info):
         # TODO: This never returns None although for realism it sometimes should
-        if column_info.choices:
+        if cat_constraint := column_info.get_constraint(Constraint.Categorical):
             # TODO: It's obviously not true in general that categories are equiprobable
-            return self.rnd.choice(column_info.choices)
+            return self.rnd.choice(cat_constraint.values)
+        elif column_info.values_used:
+            if self.rnd.randint(0, len(column_info.values_used)) != 0:
+                return self.rnd.choice(list(column_info.values_used))
```

Comment:
If you remove the call to `set()` from the `values_used` property then you don't need to transform it to a list here.